src/App.jsx

```diff
@@ -13,46 +15,90 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const SORT_BY_LENGTH = 'length';
+const SORT_BY_ALHPABET = 'alpha';
+const SET_REVERSE = 'reverse';
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const sortGoods = (arrayOfGoods, { sortValue, reverseValue }) => {
+  const copyOfGoods = [...arrayOfGoods];
+
+  const sortArray = copyOfGoods.sort((goodA, goodB) => {
+    switch (sortValue) {
+      case SORT_BY_ALHPABET:
+        return goodA.localeCompare(goodB);
+      case SORT_BY_LENGTH:
+        return goodA.length - goodB.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reverseValue) {
+    sortArray.reverse();
+  }
+
+  return sortArray;
+};
+
+export const App = () => {
+  const [arrayOfGoods, setArrayOfGoods] = useState(goodsFromServer);
```

Comment:
you don't need this state at all, every time **sortValue** or **reverseValue** changes your rendered array will change anyway, so just delete this line and pass **goodsFromServer** right into this func: 
```
sortGoods(goodsFromServer, { sortValue, reverseValue })
```